src/pull.go

```diff
@@ -36,28 +43,59 @@ func (f *PullFlags) Validate() Validations {
 
 func (f *PullOnlyFlags) Validate() Validations {
 	var validations Validations
+	if !f.hasUserAuth() && !f.hasNoUserAuth() {
+		validations = append(validations, "to authenticate, both --source-token and --source-user must be set")
+	}
 	return validations
 }
 
+func (f *PullOnlyFlags) hasUserAuth() bool {
+	return f.Username != "" && f.Token != ""
+}
+
+func (f *PullOnlyFlags) hasNoUserAuth() bool {
+	return f.Username == "" && f.Token == ""
+}
+
+func (f *PullOnlyFlags) UserAuth() (http.AuthMethod, error) {
+	if f.hasUserAuth() {
```

Comment:
I'm not sure if forcing `BasicAuth` is the right approach here. For example, the `push` command creates its client [using just a token](https://github.com/actions/actions-sync/blob/0e4f1b0152c5d8f1c36deee63e1526bbd931f7ad/src/push.go#L65-L67). It would be nice if we could make the commands consistent from an API perspective.

Additionally, I'm not sure if using `BasicAuth` would work with transports other than `http` like `ssh` or `git` which are implicitly supported. I'd have to run a few tests to see if we'd be introducing a potential regression here.